components/gantry/oneaxis/oneaxis.go

```diff
@@ -182,15 +185,92 @@ func (g *oneAxis) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 		return errors.Errorf("invalid gantry type: need 1, 2 or 0 pins per axis, have %v pins", len(newConf.LimitSwitchPins))
 	}
 
+	if g.cancelFunc != nil {
+		g.cancelFunc()
+		g.activeBackgroundWorkers.Wait()
```

Comment:
I'd put this paragraph up near the top of the function. The background worker uses `g.limitSwitchPins`, which is being redefined above here. That sounds like a potential race condition, where the background worker could read it while it's halfway through being written with new data.